src/env.cc

```diff
@@ -147,6 +147,9 @@ void Environment::Start(int argc,
 
   SetupProcessObject(this, argc, argv, exec_argc, exec_argv);
   LoadAsyncWrapperInfo(this);
+
+  CHECK_EQ(0, uv_key_create(&thread_local_env));
```

Comment:
`uv_key_create()` and `uv_key_delete()` should be called only once, not for every environment.  (The current initialization logic is probably wrong too.)

What should work is using a `uv_once_t` to protect the call to `uv_key_create()` and simply don't bother with `uv_key_delete()`.